HumanDynamicsEstimationLibrary/src/Utils.cpp

```diff
@@ -0,0 +1,398 @@
+    newState.dot_s = dot_s;
+    newState.W_p_B = W_p_B;
+    newState.dot_W_p_B = dot_W_p_B;
+    newState.W_R_B = W_R_B;
+    newState.omega_B = omega_B;
+
+    setState(newState);
+}
+
+void state::Integrator::setJointLimits(iDynTree::VectorDynSize lowerLimits,
+                                       iDynTree::VectorDynSize upperLimits,
+                                       bool active)
+{
+    if (!((lowerLimits.size() == nJoints) && (upperLimits.size() == nJoints))) {
+        yError() << LogPrefixUtils
+                 << "invalid joint limits. lowerLimits and upperLimits should have size "
+                 << nJoints;
+        return;
+    }
+
+    jointLimits.upperLimits = upperLimits;
+    jointLimits.lowerLimits = lowerLimits;
+    jointLimits.active = active;
+}
+
+void state::Integrator::getState(state& _state)
+{
+    getState(_state.s, _state.dot_s, _state.W_p_B, _state.dot_W_p_B, _state.W_R_B, _state.omega_B);
+}
+
+void state::Integrator::getState(iDynTree::VectorDynSize& s,
+                                 iDynTree::VectorDynSize& dot_s,
+                                 iDynTree::Vector3& W_p_B,
+                                 iDynTree::Vector3& dot_W_p_B,
+                                 iDynTree::Rotation& W_R_B,
+                                 iDynTree::Vector3& omega_B)
+{
+    s.resize(nJoints);
+    dot_s.resize(nJoints);
+
+    s = oldState.s;
+    dot_s = oldState.dot_s;
+    W_p_B = oldState.W_p_B;
+    dot_W_p_B = oldState.dot_W_p_B;
+    W_R_B = oldState.W_R_B;
+    omega_B = oldState.omega_B;
+}
+
+void state::Integrator::getJointState(iDynTree::VectorDynSize& s, iDynTree::VectorDynSize& dot_s)
+{
+    s.resize(nJoints);
+    dot_s.resize(nJoints);
+    s = oldState.s;
+    dot_s = oldState.dot_s;
+}
+
+void state::Integrator::getBaseState(iDynTree::Vector3& W_p_B,
+                                     iDynTree::Vector3& dot_W_p_B,
+                                     iDynTree::Rotation& W_R_B,
+                                     iDynTree::Vector3& omega_B)
+{
+    W_p_B = oldState.W_p_B;
+    dot_W_p_B = oldState.dot_W_p_B;
+    W_R_B = oldState.W_R_B;
+    omega_B = oldState.omega_B;
+}
+
+void state::Integrator::getJointConfiguration(iDynTree::VectorDynSize& s)
+{
+    s.resize(nJoints);
+    s = oldState.s;
+}
+
+void state::Integrator::getBasePose(iDynTree::Vector3& W_p_B, iDynTree::Rotation& W_R_B)
+{
+    W_p_B = oldState.W_p_B;
+    W_R_B = oldState.W_R_B;
+}
+
+void state::Integrator::getBasePose(iDynTree::Transform& W_T_B)
+{
+    iDynTree::Position basePosition;
+    iDynTree::toEigen(basePosition) = iDynTree::toEigen(oldState.W_p_B);
+    W_T_B.setPosition(basePosition);
+    W_T_B.setRotation(oldState.W_R_B);
+}
+
+void state::Integrator::integrate(iDynTree::VectorDynSize new_dot_s, double dt)
+{
+    if (!(new_dot_s.size() == nJoints)) {
+        yError() << LogPrefixUtils << "invalid dot_s. dot_s expected to have size " << nJoints;
+    }
+
+    if (interpolator == rectangular) {
```

Comment:
Convert to switch-case without `default` case? It has the advantage of generating a warning in the case a new case is added in the future.